public/pages/HistoricalDetectorDetail/containers/HistoricalDetectorDetail/HistoricalDetectorDetail.tsx

```diff
@@ -0,0 +1,410 @@
+      ]);
+    }
+  }, [detector]);
+
+  const fetchDetector = async () => {
+    try {
+      await dispatch(getDetector(detectorId));
+      setIsLoadingDetector(false);
+    } catch {
+      core.notifications.toasts.addDanger(
+        'Error retrieving the historical detector'
+      );
+    }
+  };
+
+  // Try to get the detector initially
+  useEffect(() => {
+    if (detectorId) {
+      setIsLoadingDetector(true);
+      fetchDetector();
+    }
+  }, []);
+
+  // If detector is initialiazing or running: keep fetching every second to quickly update state/results/percentage bar, etc.
+  useEffect(() => {
+    if (
+      detector?.curState === DETECTOR_STATE.RUNNING ||
+      detector?.curState === DETECTOR_STATE.INIT
+    ) {
+      const intervalId = setInterval(
+        fetchDetector,
+        HISTORICAL_DETECTOR_RESULT_REFRESH_RATE
+      );
+      return () => {
+        clearInterval(intervalId);
+      };
+    }
+  }, [detector]);
+
+  const handleDeleteAction = () => {
+    setHistoricalDetectorDetailModalState({
+      ...historicalDetectorDetailModalState,
+      isOpen: true,
+      action: HISTORICAL_DETECTOR_ACTION.DELETE,
+    });
+  };
+
+  const handleEditAction = () => {
+    detector?.curState === DETECTOR_STATE.RUNNING ||
+    detector?.curState === DETECTOR_STATE.INIT
+      ? setHistoricalDetectorDetailModalState({
+          ...historicalDetectorDetailModalState,
+          isOpen: true,
+          action: HISTORICAL_DETECTOR_ACTION.EDIT,
+        })
+      : props.history.push(`/historical-detectors/${detectorId}/edit`);
+  };
+
+  const onStartDetector = async () => {
+    try {
+      // Check to make sure we can start. Current historical detector limit: 2 running per node
+      const clusterStatsResponse = await dispatch(getClusterStats());
+      const nodeCount = get(
+        clusterStatsResponse,
+        'response.nodes.count.data',
+        1
+      );
+      const historicalDetectorsResponse = await dispatch(
+        getHistoricalDetectorList(GET_ALL_DETECTORS_QUERY_PARAMS)
+      );
+      const runningCount = getRunningHistoricalDetectorCount(
+        get(historicalDetectorsResponse, 'response.detectorList')
+      );
+
+      if (nodeCount * 2 <= runningCount) {
+        throw 'only 2 historical detectors can be running per data node in the cluster';
+      }
+
+      await dispatch(startDetector(detectorId));
+      fetchDetector();
+      core.notifications.toasts.addSuccess(
+        `Successfully started the historical detector`
+      );
+    } catch (err) {
+      core.notifications.toasts.addDanger(
+        `There was a problem starting the historical detector: ${err}`
+      );
+    }
+  };
+
+  // We query the task state 5s after making the stop detector call. If the task is still running,
+  // then it is assumed there was an error stopping this task / historical detector.
+  const onStopDetector = async (isDelete: boolean, listener?: Listener) => {
```

Comment:
`isDelete` seems not so readable, how about we use `HISTORICAL_DETECTOR_ACTION` here ? If need to stop detector for delete action, we can pass `HISTORICAL_DETECTOR_ACTION.DELETE`

Reply:
I think the idea of passing a `HISTORICAL_DETECTOR_ACTION` here makes much more sense, will change to that.